frb_codegen/src/generator/rust/ty_enum.rs

```diff
@@ -276,12 +277,32 @@ impl TypeRustGeneratorTrait for TypeEnumRefGenerator<'_> {
         )
     }
 
+    fn deallocate_funcs(
```

Comment:
As for the Rust impl of dealloc func, is it possible that, we somehow reuse the code generator for a normal func? Because "a normal func accepting type T" is indeed properly deallocating the value of type T.

Otherwise, now we are writing dealloc for each and every type, which looks like duplication, and is also quite hard to test.

Reply:
I made a prototype but `wire2api -> Result<T, String>`

then we could do wire2api for all arguments, fields and subfields. And call unwrap already on the executor.
Then the memory will be safely cleared for all fields.

```
impl Wire2Api<OpaqueNested> for wire_OpaqueNested {
    fn wire2api(self) -> Result<OpaqueNested, &'static str> {
        Ok({
            let field0 = self.first.wire2api();
            let field1 = self.second.wire2api();
            OpaqueNested {
                first: field0?,
                second: field1?,
            }
        })
    }
}


fn wire_run_nested_opaque_impl(
    port_: MessagePort,
    opaque: impl Wire2Api<OpaqueNested> + UnwindSafe,
) {
    FLUTTER_RUST_BRIDGE_HANDLER.wrap(
        WrapInfo {
            debug_name: "run_nested_opaque",
            port: Some(port_),
            mode: FfiCallMode::Normal,
        },
        move || {
            let api_opaque = opaque.wire2api();
            move |task_callback| Ok(run_nested_opaque(api_opaque.unwrap()))
        },
    )
}
```


in such an implementation, each field and argument will be prepared and, in case of a panic on the executor, will be cleared

Reply:
The prototype idea LGTM. Not sure, but maybe we do not need to be that heavy, i.e. no need to HANDLER.wrap?

Btw, not sure whether `first: field0?` is the correct way. Maybe we should do `field0 = wire2api()?` instead, i.e. propagate error early?

Reply:
I didn't quite understand about `HANDLER.wrap`.
An early return is not allowed, since we must call `wire2api` for all fields, otherwise leak pointers

Reply:
> I didn't quite understand about HANDLER.wrap

Look at its source code which IIRC is not long

> An early return is not allowed, since we must call wire2api for all fields, otherwise leak pointers

Nice catch! Btw may be great to mention it somewhere in code comments for future readers

Reply:
> What about looking at its source code which IIRC is not long. IIRC, it is about catching errors etc

You are talking about replacing panic on unwrap with Error?

```
move |task_callback| Ok(run_nested_opaque(api_opaque.map_err(|e| anyhow::anyhow!(e))?))
```

Reply:
Well I did not mean that. Looking at https://github.com/fzyzcjy/flutter_rust_bridge/blob/265ee91038056092954ff6fa733fd05693b107bd/frb_rust/src/handler.rs#L95, IMHO maybe we can do sth like (following is a rough draft):

```
fn wire_eat_opaque_object_and_indeed_do_nothing(
    opaque: impl Wire2Api<OpaqueNested> + UnwindSafe,
) {
    catch_unwind_related_things(|| {
      opaque.wire2api().unwrap();
    });
}
```

since we do not really need those WrapInfo etc

Reply:
maybe later?
I would not like to climb into the handler as part of this PR

Reply:
Sure, then leave it later